data/ha/hawk_test.py

```diff
@@ -0,0 +1,829 @@
+            'remove_clone', 'add_group', 'remove_group', 'click_around_edit_conf']
+
+# XPATH constants
+CLICK_OK_SUBMIT = "//*[@id=\"modal\"]/div/div/form/div[3]/input"
+CONFIG_EDIT = '//a[contains(@href, "config/edit")]'
+OCF_OPT_LIST = '//option[contains(@value, "ocf")]'
+ANYTHING_OPT_LIST = '//option[contains(@value, "anything")]'
+EDIT_START_TIMEOUT = "//*[@id=\"oplist\"]/fieldset/div/div[1]/div[1]/div[2]/div/div/a[1]"
+EDIT_STOP_TIMEOUT = "//*[@id=\"oplist\"]/fieldset/div/div[1]/div[2]/div[2]/div/div/a[1]"
+MODAL_TIMEOUT = "//*[@id=\"modal\"]/div/div/form/div[2]/fieldset/div/div[1]/div/div"
+MODAL_STOP = "//*[@id=\"modal\"]/div/div/form/div[2]/fieldset/div/div[2]/div/div/select/option[6]"
+EDIT_MONITOR_TIMEOUT = "//*[@id=\"oplist\"]/fieldset/div/div[1]/div[3]/div[2]/div/div/a[1]"
+MODAL_MONITOR_TIMEOUT = "//*[@id=\"modal\"]/div/div/form/div[2]/fieldset/div/div[1]/div"
+TARGET_ROLE_STARTED = '//option[contains(@value, "tarted")]'
+HREF_DELETE_FORMAT = '//a[contains(@href, "%s") and contains(@title, "Delete")]'
+COMMIT_BTN_DANGER = '//button[contains(@class, "btn-danger") and contains(@class, "commit")]'
+CLONE_DATA_HELP_FILTER = '//a[contains(@data-help-filter, ".clone")]'
+OPT_STONITH = '//option[contains(@value, "stonith-sbd")]'
+RSC_OK_SUBMIT = '//input[contains(@class, "submit")]'
+GROUP_DATA_FILTER = '//a[contains(@data-help-filter, ".group")]'
+STONITH_CHKBOX = '//input[contains(@type, "checkbox") and contains(@value, "stonith-sbd")]'
+HREF_CONSTRAINTS = '//a[contains(@href, "#constraints")]'
+HREF_NODES = '//a[contains(@href, "#nodes")]'
+HREF_TAGS = '//a[contains(@href, "#tags")]'
+HREF_ALERTS = '//a[contains(@href, "#alerts")]'
+HREF_FENCING = '//a[contains(@href, "#fencing")]'
+RSC_ROWS = "//*[@id=\"resources\"]/div[1]/div[2]/div[2]/table/tbody/tr"
+DROP_DOWN_FORMAT = "//*[@id=\"resources\"]/div[1]/div[2]/div[2]/table/tbody/tr[%d]/td[6]/div/div"
+STONITH_MAINT_ON = '//a[contains(@href, "stonith-sbd/maintenance_on")]'
+NODE_DETAILS = "//*[@id=\"nodes\"]/div[1]/div[2]/div[2]/table/tbody/tr[1]/td[5]/div/a[2]"
+DISMISS_MODAL = "//*[@id=\"modal\"]/div/div/div[3]/button"
+CLEAR_STATE = "//*[@id=\"nodes\"]/div[1]/div[2]/div[2]/table/tbody/tr[1]/td[5]/div/div/button"
+NODE_MAINT = '//a[contains(@href, "maintenance") and contains(@title, "Switch to maintenance")]'
+NODE_READY = '//a[contains(@href, "ready") and contains(@title, "Switch to ready")]'
+GENERATE_REPORT = "//*[@id=\"generate\"]/form/div/div[2]/button"
+CLONE_CHILD = ('//select[contains(@data-help-filter, ".row.resource") and '
+               'contains(@name, "clone[child]")]')
+TARGET_ROLE_FORMAT = ('//select[contains(@class, "form-control") and '
+                      'contains(@name, "%s[meta][target-role]")]')
+STONITH_MAINT_OFF = ('//a[contains(@href, "stonith-sbd") and '
+                     'contains(@title, "Disable Maintenance Mode")]')
+
+# Functions
+
+### Functions to check types and classes
+
+def checkint(i):
```

Comment:
the same would be with checkint and checkstr functions I would prefer to get exception instead hiding the fact that I am passing "pinapple" to function which expect some integer

Reply:
That's precisely the thing: I am trying to avoid exceptions whenever possible. When there's an exception, the script aborts completely; if this happens after the browser has already been opened, the browser remains open without a controlling driver because it does not reach the call to `browser.quit()`. I guess I could add a call to quit the browser when `except Exception:`, but figured it was not a good practice.